docs/deployment/gae.rst

```diff
@@ -1,225 +1,104 @@
 
-   .. code-block:: text
-      :linenos:
+Setup
+-----
 
-      $ cd ../..
-      $ python2.5 /usr/local/bin/dev_appserver.py pyramidapp/app/
+   First we'll need to create a few files so that App Engine can communicate with our project properly.
 
-   Startup success looks something like this:
+   Create a ``requirements.txt`` file, a ``main.py`` file, an ``app.yaml`` file, and an ``appengine_config.py`` file.
 
-   .. code-block:: text
+#. Edit ``requirements.txt``
 
-      [chrism@vitaminf pyramid_gae]$ python2.5 \
-                    /usr/local/bin/dev_appserver.py \
-                    pyramidapp/app/
-      INFO     2009-05-03 22:23:13,887 appengine_rpc.py:157] # ... more... 
-      Running application pyramidapp on port 8080: http://localhost:8080
+    Add the following lines:
 
-   You may need to run "Make Symlinks" from the Google App Engine
-   Launcher GUI application if your system doesn't already have the
-   ``dev_appserver.py`` script sitting around somewhere.
+    .. code-block:: text
+        :linenos:
 
-#. Hack on your pyramid application, using a normal run, debug, restart
-   process.  For tips on how to use the ``pdb`` module within Google
-   App Engine, `see this blog post
-   <http://jjinux.blogspot.com/2008/05/python-debugging-google-app-engine-apps.html>`_.
-   In particular, you can create a function like so and call it to
-   drop your console into a pdb trace::
+        Pyramid
+        waitress
+        pyramid_debugtoolbar
+        pyramid_chameleon
 
-      def set_trace():
-          import pdb, sys
-          debugger = pdb.Pdb(stdin=sys.__stdin__, 
-              stdout=sys.__stdout__)
-          debugger.set_trace(sys._getframe().f_back)
+#. Edit ``main.py``
 
-#. `Sign up for a GAE account <http://code.google.com/appengine/>`_
-   and create an application.  You'll need a mobile phone to accept an
-   SMS in order to receive authorization.
+   Add the following lines::
 
-#. Edit the application's ID in ``app.yaml`` to match the application
-   name you created during GAE account setup.
+        from pyramid.paster import get_app, setup_logging
+        ini_path = 'production.ini'
+        setup_logging(ini_path)
+        application = get_app(ini_path, 'main')
 
-   .. code-block:: yaml
+#. Edit ``appengine_config.py``
 
-      application: mycoolpyramidapp
+   Add the following lines::
 
-#. Upload the application
+        from google.appengine.ext import vendor
+        vendor.add('lib')
 
-   .. code-block:: text
+#. Edit ``app.yaml``
 
-      $ python2.5 /usr/local/bin/appcfg.py update pyramidapp/app
+   Add the following lines:
 
-   You almost certainly won't hit the 3000-file GAE file number limit
-   when invoking this command.  If you do, however, it will look like
-   so:
+    .. code-block:: yaml
 
-   .. code-block:: text
+        application: application-id
+        version: version
+        runtime: python27
+        api_version: 1
+        threadsafe: false
```

Comment:
You might need to adjust the indentation here. The "a" in "application:" should be directly below the first "o" in ".. code-block::".